src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java

```diff
@@ -0,0 +1,122 @@
+
+        SearchEventAndCase searchEventAndCase = new SearchEventAndCase(task.get("caseId"), "requestRespondentEvidence");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchEventAndCase,
+            authorizationHeadersProvider.getLawFirmAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.FORBIDDEN.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.FORBIDDEN.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.FORBIDDEN.value()))
+            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
+    }
+
+    @Test
+    public void should_return_a_200_and_retrieve_a_task_by_event_and_case_location_match() {
+
+        Map<CamundaVariableDefinition, String> variablesOverride = Map.of(
+            CamundaVariableDefinition.JURISDICTION, "IA",
+            CamundaVariableDefinition.LOCATION, "765324",
+            CamundaVariableDefinition.TYPE, "ReviewTheAppeal",
+            CamundaVariableDefinition.TASK_ID, "ReviewTheAppeal",
+            CamundaVariableDefinition.TASK_STATE, "unassigned"
+        );
+
+        Map<String, String> task = common.setupTaskAndRetrieveIdsWithCustomVariablesOverride(variablesOverride);
+
+        SearchEventAndCase searchEventAndCase = new SearchEventAndCase(task.get("caseId"), "requestRespondentEvidence");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchEventAndCase,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("tasks[0].task_state", equalTo("unassigned"))
+            .body("tasks[0].case_id", equalTo(task.get("caseId")))
+            .body("tasks[0].id", equalTo(task.get("taskId")))
+            .body("tasks[0].type", equalTo("ReviewTheAppeal"));
+
+
+    }
+
+    @Test
+    public void should_return_a_200_and_retrieve_a_task_by_id_jurisdiction_location_and_region_match() {
+
+        Map<String, String> task = common.setupTaskAndRetrieveIds();
+
+        SearchEventAndCase searchEventAndCase = new SearchEventAndCase(task.get("caseId"), "requestRespondentEvidence");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            searchEventAndCase,
+            authorizationHeadersProvider.getTribunalCaseworkerBAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("tasks[0]", equalTo(null));
+    }
+
```

Comment:
There are 3 other AC's that are not covered. Can this be added to the functional test;
Given the consuming service is completing an event,
And the consuming service provides the ID of the event that is being completed and the case ID;

AC1
And the event ID matches a row (or rows) in the Task Completion DMN table
And the matching task types are found in the active tasks for that Case
Then return a list of tasks that are completed by that event
And this list should include of the standard attributes as defined above 

AC2
And the event ID does not match any row in the Task Completion DMN table
Then return an empty response, indicating no matches found

AC3
And the event ID matches a row (or rows) in the Task Completion DMN table
And the matching task types are NOT found in the active tasks for that Case
Then return an empty response, indicating no matches found